contracts/src/rollup/RollupLib.sol

```diff
@@ -19,21 +19,21 @@ library RollupLib {
     using GlobalStateLib for GlobalState;
 
     // Not the same as a machine hash for a given execution state
-    function executionStateHash(ExecutionState memory state) internal pure returns (bytes32) {
+    function assertionStateHash(AssertionState memory state) internal pure returns (bytes32) {
         return keccak256(abi.encodePacked(state.machineStatus, state.globalState.hash()));
```

Comment:
should `state.endHistoryRoot` be included in the hashing?

Reply:
After looking at this deeper, I think we should include it @gzeoneth

Reply:
Nice catch @godzillaba

Reply:
fixed in ff48288